packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -3118,9 +3123,9 @@ class SemanticsOwner extends ChangeNotifier {
       final CustomSemanticsAction action = CustomSemanticsAction.getAction(actionId)!;
       builder.updateCustomAction(id: actionId, label: action.label, hint: action.hint, overrideId: action.action?.index ?? -1);
     }
-    // TODO(a-wallen): https://github.com/flutter/flutter/issues/112221
-    // ignore: deprecated_member_use
-    SemanticsBinding.instance.platformDispatcher.updateSemantics(builder.build());
+    if(onSemanticsUpdate != null) {
```

Comment:
nit: add space after `if`

Reply:
I think I would assert here if onSemanticsUpdate is null. It is almost certainly a bug when `sendSemanticsUpdate` is called and onSemanticsUpdate wasn't configured.

Reply:
Actually, onSemanticsUpdate should be required on the SemanticsOwner. There is no point in creating a semantics owner if this is not provided.

Reply:
(On the PipelineOwner, configuring this should be optional, though. It is ok to use a PipelineOwner and never request semantics. So we should probably just assert for the PipelineOwner if you're trying to turn on semantics and onSemanticsUpdate is not configured)